### PR TITLE
feat: Internal only datasets

### PR DESCRIPTION
**Pull request by lynnagara**
Provide a way to register datasets as internal only. This prevents them
from being queried directly via the API by Sentry (and the web UI by
extension). The internal datasets can still be queried indirectly.

This is specifically for the (future) querylog dataset but it might also
apply to other datasets.

### CODE REVIEW

File: snuba/datasets/factory.py [hunk 1]
```diff
@@ -19,6 +19,8 @@
     "discover",
 }
 
+# Internal datasets registered here cannot be queried directly via API
+INTERNAL_DATASET_NAMES: Set[str] = {}
```
[the remark this answers is not in the document]
Comment (lynnagara) [reply snuba/datasets/factory.py]: Oops, thanks

**Review by fpacifici — COMMENTED** [review]
I think the DatasetConverter is not the right place for this check:
1) it is not strict enough since we can still pass a query to /query and the dataset name as POST parameter. That will use the bare get_dataset function and bypass this check
2) if we put the check in get_dataset it will become too strict since you won't be able to load the dataset in the consumer.

**Comment by lynnagara** [comment]
Damn, forgot this wasn't part of the URL for POST

**Comment by lynnagara** [comment]
Moved into dataset_query, this is just enforced for actual POST queries now

**Comment by lynnagara** [comment]
Yeah, this is the most basic version possible. The only motivation for this right now is to block Sentry inadvertently adding tomorrow a product or feature that uses the querylog dataset since it's not appropriate for production use, without Snuba at least knowing about it.

**Review by tkaemming — APPROVED** [review]

**Pull request merged**